lfs/pointer.go

```diff
@@ -89,48 +111,106 @@ func verifyVersion(version string) error {
 }
 
 func decodeKV(data []byte) (*Pointer, error) {
-	parsed, err := decodeKVData(data)
+	kvps, exts, err := decodeKVData(data)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := verifyVersion(parsed["version"]); err != nil {
+	if err := verifyVersion(kvps["version"]); err != nil {
 		return nil, err
 	}
 
-	oidValue, ok := parsed["oid"]
+	value, ok := kvps["oid"]
 	if !ok {
 		return nil, errors.New("Invalid Oid")
 	}
 
-	oidParts := strings.SplitN(oidValue, ":", 2)
-	if len(oidParts) != 2 {
-		return nil, errors.New("Invalid Oid type in" + oidValue)
-	}
-	if oidParts[0] != oidType {
-		return nil, errors.New("Invalid Oid type: " + oidParts[0])
+	oid, err := parseOid(value)
+	if err != nil {
+		return nil, err
 	}
-	oid := oidParts[1]
 
-	var size int64
-	sizeStr, ok := parsed["size"]
+	value, ok = kvps["size"]
 	if !ok {
-		return nil, errors.New("Invalid Oid")
+		return nil, errors.New("Invalid size")
+	}
```

Comment:
I think we can skip checking if `kvps` has a `size` key: http://play.golang.org/p/ij4GbJITJ4

```go
value = kvps["size"]
size, err := strconv.ParseInt(value, 10, 0)
if err != nil || size < 0 {
  return nil, fmt.Errorf("Invalid size: %q", value)
}
```